src/class/Socket/Socket.cpp

```diff
@@ -0,0 +1,145 @@
+
+#include <Socket/Socket.hpp>
+
+/*
+** ------------------------------- CONSTRUCTOR --------------------------------
+*/
+
+Socket::Socket(IP ip, Port port)
+	: _ip(ip)
+	, _port(port)
+	, _binded(false)
+	, _fd(-1)
+	, _domain(AF_INET)
+	, _type(SOCK_STREAM)
+	, _protocol(0)
+{
+	/* Create socket */
+	this->_fd = socket(this->_domain, this->_type, this->_protocol);
+	if (this->_fd <= -1)
+		throw SocketCreationFailed();
+
+	/* Set the socket to reuse (makes it re-bindable directly after close) */
+	int reuse_addr = true;
+	setsockopt(this->_fd, SOL_SOCKET, SO_REUSEADDR, &reuse_addr, sizeof(reuse_addr));
+
+	/* Init sockaddr */
+	memset(&this->_sockaddr, 0, sizeof(this->_sockaddr));
+	this->_sockaddr.sin_family = this->_domain;
+	this->_sockaddr.sin_port = htons(this->_port.get());
+	this->_sockaddr.sin_addr.s_addr = inet_addr(this->_ip.get().c_str());
+}
+
+Socket::Socket(Socket const &src)
+	: _ip(src._ip)
+	, _port(src._port)
+	, _binded(src._binded)
+	, _fd(src._fd)
+	, _domain(src._domain)
+	, _type(src._type)
+	, _protocol(src._protocol)
+	, _sockaddr(src._sockaddr)
+{
+}
+
+const char *Socket::SocketCreationFailed::what() const throw()
+{
+	return "Socket creation failed";
+}
+const char *Socket::SocketListeningFailed::what() const throw()
+{
+	return "Socket setting to listening mode failed";
+}
+const char *Socket::SocketBindingFailed::what() const throw()
+{
+	return "Socket binding failed";
+}
+const char *Socket::SocketNotBinded::what() const throw()
+{
+	return "Socket not binded";
+}
+
+/*
+** -------------------------------- DESTRUCTOR --------------------------------
+*/
+
+Socket::~Socket()
```

Comment:
no close() here?
i just want to make sure we are closing it somewhere else

Reply:
It is the server which uses the socket close function, should I just place it in the constructor?

Reply:
if it is possible to close on dtor i think it is more readable.